ui/src/message_popup/mobile_prototype/texting_interface.jsx

```diff
@@ -0,0 +1,184 @@
+  },
+  
+  onRevisionSendMessage(){
+    this.props.setRevisedResponse(this.state.text);
+    this.props.log.logRevision();
+  },
+  
+  onPassSendMessage(){
+    this.props.setPassedResponse('');
+    this.props.log.logRevisionDeclined();
+  },
+  
+  onNextQuestion(){
+    this.props.onQuestionDone();
+  },
+  
+  render(){
+    const mainButtonLabel = this.props.isReadyToMoveOn() ? this.props.nextButtonLabel : this.props.helpType !== 'feedback' ? 'Send' : this.state.initialMessageSent ? 'Revise Response' : 'Save Response';
+    return (
+      <div>
+        <Divider />
+        <div style={styles.responseTextBox}>
+          <TextField 
+            id="Response"
+            multiLine={true}
+            underlineShow={false}
+            rows={2}
+            rowsMax={2}
+            onChange={this.onTextChange}
+            disabled={this.props.isReadyToMoveOn()}
+            value={this.state.text === undefined ? '' : this.state.text}
+            hintText="Type something you would say"
+            style={styles.responseTextField}
+            textareaStyle={styles.responseTextAreaInner}
+            />
+        </div>
+        <div style={styles.responseButtonRow}>  
+          <RaisedButton
+            style={styles.responseButton}
+            label={mainButtonLabel}
+            primary={this.props.isReadyToMoveOn()}
+            secondary={!this.props.isReadyToMoveOn()}
+            onTouchTap={this.props.isReadyToMoveOn() ? this.onNextQuestion : this.state.initialMessageSent ?  this.onRevisionSendMessage : this.onSendMessage}
+            disabled={(this.state.text === undefined || this.state.text === '') && !this.props.isReadyToMoveOn()}
+            />
+          {this.props.helpType === 'feedback' && this.state.initialMessageSent && !this.props.isReadyToMoveOn() &&
+          <RaisedButton
+            style={styles.responseButton}
+            label="Pass"
+            onTouchTap={this.onPassSendMessage}
+            />
+          }
+          {this.props.helpType === 'hints' && !this.props.isReadyToMoveOn() &&
+          <RaisedButton
+            style={styles.responseButton}
+            label="Show Example"
+            disabled={this.props.nextExample(false) === null}
+            onTouchTap={this.props.showExample}
+            />
+          }
+          <div style={styles.responseTimer}>
+            {this.props.elapsedMs/1000 + 's'}
+          </div>
+        </div>
+      </div>
+    );
+  }
+});
+
+const styles = {
+  responseTextBox: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-between'
+  },
+  responseTextField: {
+    width: '100%',
+    paddingLeft: 20,
+    paddingRight: 20,
+  },
+  responseTextAreaInner: {
+    border: '1px solid #eee',
+    marginBottom: 0
+  },
+  responseButtonRow: {
+    marginTop: 20,
+    marginLeft: 15,
+    marginRight: 15,
+  },
+  responseButton: {
+    marginLeft: 5,
+    marginRight: 5
+  },
+  responseTimer: {
+    position: 'fixed',
```

Comment:
Can you use flexbox here?  I don't think we want fixed position for anything if we can avoid it.